src/core/verfer.rs

```diff
@@ -0,0 +1,246 @@
+use crate::core::matter::{tables as matter, Matter};
+use crate::error::{Error, Result};
+
+trait Verfer {
+    fn new_with_code_and_raw(code: &str, raw: &[u8]) -> Result<Matter>
+    where
+        Self: Sized;
+    fn new_with_qb64(qb64: &str) -> Result<Matter>
+    where
+        Self: Sized;
+    fn new_with_qb64b(qb64b: &[u8]) -> Result<Matter>
+    where
+        Self: Sized;
+    fn new_with_qb2(qb2: &[u8]) -> Result<Matter>
+    where
+        Self: Sized;
+    fn verify(&self, sig: &[u8], ser: &[u8]) -> Result<bool>;
+}
+
+fn validate_code(code: &str) -> bool {
+    vec![
+        matter::Codex::Ed25519N.code(),
+        matter::Codex::Ed25519.code(),
+        matter::Codex::ECDSA_256k1N.code(),
+        matter::Codex::ECDSA_256k1.code(),
+        // matter::Codex::Ed448N.code(),
```

Comment:
I'll create an issue for the ed448 implementation and link it to here, and just remove the comments.